pkg/operator/controller/status.go

```diff
@@ -95,123 +99,184 @@ func (r *reconciler) syncOperatorStatus() error {
 
+	return setOperatorLastTransitionTime(progressingCondition, oldCondition)
+}
+
+// computeOperatorAvailableCondition computes the operator's current Available status state.
+func computeOperatorAvailableCondition(oldCondition *configv1.ClusterOperatorStatusCondition,
+	anyDNSesAvailable bool) configv1.ClusterOperatorStatusCondition {
 	availableCondition := &configv1.ClusterOperatorStatusCondition{
-		Type:   configv1.OperatorAvailable,
-		Status: configv1.ConditionUnknown,
-	}
-	daemonsetsAvailable := map[string]bool{}
-	for _, d := range daemonsets {
-		daemonsetsAvailable[d.Name] = d.Status.NumberAvailable > 0
-	}
-	unavailable := []string{}
-	for _, dns := range dnses {
-		name := DNSDaemonSetName(&dns).Name
-		if available, exists := daemonsetsAvailable[name]; !exists {
-			msg := fmt.Sprintf("no daemonset for dns %q", dns.Name)
-			unavailable = append(unavailable, msg)
-		} else if !available {
-			msg := fmt.Sprintf("daemonset %q is not available", name)
-			unavailable = append(unavailable, msg)
-		}
+		Type: configv1.OperatorAvailable,
 	}
-	if len(unavailable) == 0 {
+	if anyDNSesAvailable {
 		availableCondition.Status = configv1.ConditionTrue
+		availableCondition.Reason = "AsExpected"
+		availableCondition.Message = "At least 1 dns resource is available"
 	} else {
 		availableCondition.Status = configv1.ConditionFalse
-		availableCondition.Reason = "DaemonSetNotAvailable"
-		availableCondition.Message = strings.Join(unavailable, "\n")
+		availableCondition.Reason = "DNSUnavailable"
+		availableCondition.Message = fmt.Sprintf("No dns resources available")
 	}
-	conditions = setStatusCondition(conditions, availableCondition)
 
-	return conditions
+	return setOperatorLastTransitionTime(availableCondition, oldCondition)
 }
 
-// setStatusCondition returns the result of setting the specified condition in
-// the given slice of conditions.
-func setStatusCondition(oldConditions []configv1.ClusterOperatorStatusCondition, condition *configv1.ClusterOperatorStatusCondition) []configv1.ClusterOperatorStatusCondition {
-	condition.LastTransitionTime = metav1.Now()
+// checkAllDNSesAvailable checks if all DNS resources are available.
+func checkAllDNSesAvailable(dnses []operatorv1.DNS) bool {
+	for _, dns := range dnses {
+		available := false
+		for _, c := range dns.Status.Conditions {
+			if c.Type == operatorv1.OperatorStatusTypeAvailable && c.Status == operatorv1.ConditionTrue {
+				available = true
+				break
+			}
+		}
+		if !available {
+			return false
+		}
+	}
 
-	newConditions := []configv1.ClusterOperatorStatusCondition{}
+	return true
+}
 
-	found := false
-	for _, c := range oldConditions {
-		if condition.Type == c.Type {
-			if condition.Status == c.Status &&
-				condition.Reason == c.Reason &&
-				condition.Message == c.Message {
-				return oldConditions
+// checkAnyDNSesAvailable checks if all DNS resources are available.
+func checkAnyDNSesAvailable(dnses []operatorv1.DNS) bool {
+	var dnsesAvailable int
```

Comment:
Note for follow-up: This can be a Boolean value.  (Alternatively, if we decide not to combine `checkAnyDNSesAvailable` and `checkAllDNSesAvailable` as I suggested in another comment, we can do away with the variable and just `dnsesAvailable++`→`return true` below.)